applications/conversations/controllers/api/MessagesApiController.php

```diff
@@ -0,0 +1,370 @@
+     */
+    private function fullSchema() {
+        /** @var Schema $schema */
+        static $schema;
+
+        if ($schema === null) {
+            // Name this schema so that it can be read by swagger.
+            $schema = $this->schema([
+                'messageID:i' => 'The ID of the message.',
+                'conversationID:i' => 'The ID of the conversation.',
+                'body:s' => [
+                    'maxLength' => $this->config->get('MaxLength', 2000),
+                    'description' => 'The body of the message.',
+                ],
+                'insertUserID:i' => 'The user that created the message.',
+                'insertUser?' => $this->getUserFragmentSchema(),
+                'dateInserted:dt' => 'When the message was created.',
+            ], 'Message');
+        }
+
+        return $schema;
+    }
+
+    /**
+     * Get a message.
+     *
+     * @param int $id The ID of the message.
+     * @throws NotFoundException if the message could not be found.
+     * @return array
+     */
+    public function get($id) {
+        $this->permission('Conversations.Conversations.Add');
+
+        $this->schema([
+            'id:i' => 'The message ID.'
+        ], 'in')->setDescription('Get a message.');
+        $out = $this->schema($this->fullSchema(), 'out');
+
+        $message = $this->messageByID($id);
+
+        $isInConversation = $this->conversationModel->inConversation($message['ConversationID'], $this->getSession()->UserID);
+
+        if (!$isInConversation) {
+            $this->checkModerationPermission();
+        }
+
+        $this->userModel->expandUsers($message, ['InsertUserID']);
+
+        $message = $this->prepareRow($message);
+        return $out->validate($message);
+    }
+
+    /**
+     * List messages of a user.
+     *
+     * @param array $query The query string.
+     * @return array
+     */
+    public function index(array $query) {
+        $this->permission('Conversations.Conversations.Add');
+
+        $in = $this->schema([
+                'conversationID:i?'=> 'Filter messages by conversation.',
+                'insertUserID:i?' => 'Filter messages by specified user. '.
+                    'Use in conjunction with conversationID to have all messages of a conversation that were created by a specific user.',
+                'page:i?' => [
+                    'description' => 'Page number.',
+                    'default' => 1,
+                    'minimum' => 1,
+                ],
+                'limit:i?' => [
+                    'description' => 'The number of items per page.',
+                    'default' => $this->config->get('Conversations.Messages.PerPage', 50),
+                    'minimum' => 1,
+                    'maximum' => 100
+                ],
+                'expand:b?' => 'Expand associated records.'
+            ], 'in')
+            ->setDescription('List user messages. By default, get the messages list of the current users.');
```

Comment:
> List user messages. By default, get the messages list of the current users.

Nothing is required.

Reply:
> [...] instead of forcing filtering by the current user, later on.

I get that nothing is required. The default behavior is to show me a bunch of messages I've sent, out of context? That doesn't seem very intuitive to me. If I want that, it seems like I should have to specifically request that. This is similar to the comments index. There, you have to specify a discussion ID *or* the ID of a user you want to query comments for.

Reply:
The problem is that Messages are a bit different because they are not public.
So you are essentially filtering by insertUserID by default.

The parameter insertUserID is only there for moderators.

This combined with pagination let you get all  your messages without requiring to go on a per conversation basis. Maybe @tburry could give his 2cent on this. I don't know what is the correct way of doing it. 

I do see that it could make sense to require a conversationID but this is limiting.

Reply:
After a talk with Ryan, long story short: We could either return everything that you have access to or require that you filter by something. It seems that for comments we already choose to require filtering so I'm gonna do the same thing there.